bindings/nodejs/src/lib.rs

```diff
@@ -50,66 +78,189 @@ pub struct ObjectMeta {
 }
 
 #[napi]
-pub struct Operator {
-    inner: opendal::Operator
+pub enum ObjectMode {
+    /// FILE means the object has data to read.
+    FILE,
+    /// DIR means the object can be listed.
+    DIR,
+    /// Unknown means we don't know what we can do on this object.
+    Unknown,
 }
 
+#[allow(dead_code)]
 #[napi]
-impl Operator {
-    pub fn new(op: opendal::Operator) -> Self {
+pub struct ObjectMetadata {
+    /// Mode of this object.
+    pub mode: ObjectMode,
+
+    /// Content-Disposition of this object
+    pub content_disposition: Option<String>,
+    /// Content Length of this object
+    pub content_length: Option<u32>,
+    /// Content MD5 of this object.
+    pub content_md5: Option<String>,
+    /// Content Range of this object.
+    pub content_range: Option<Vec<u32>>,
+    /// Content Type of this object.
+    pub content_type: Option<String>,
+    /// ETag of this object.
+    pub etag: Option<String>,
+    /// Last Modified of this object.
+    pub last_modified: i64,
+}
+
+#[napi]
+pub struct ObjectLister {
+    inner: opendal::ObjectLister
+}
+
+#[napi]
+impl ObjectLister {
+    pub fn new(op: opendal::ObjectLister) -> Self {
         Self { inner: op }
     }
 
     #[napi]
-    pub async fn meta(&self, path: String) -> Result<ObjectMeta> {
-        let o = self.inner.object(&path);
-        let meta = o
+    pub async unsafe fn next_page(&mut self) -> Result<Vec<DataObject>> {
+        Ok(self.inner
+            .next_page()
+            .await
+            .map_err(format_napi_error)?.unwrap_or_default()
+            .iter()
+            .map(|obj| DataObject { inner: obj.to_owned()})
+            .collect())
+    }
+}
+
+#[napi]
+pub struct DataObject {
```

Comment:
How about using `Object(opendal::Object)` instead?

Reply:
Is this pattern popular in developing rust? I'm confused about this.
Using 0 to find the first prop is a little strange thing to me.

Reply:
For more references:

- https://doc.rust-lang.org/rust-by-example/generics/new_types.html
- https://rust-unofficial.github.io/patterns/patterns/behavioural/newtype.html

Reply:
LGTM.